src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -481,14 +479,17 @@ def geop_tasks(geom, errback, exchange, choose_worker):
          nlcd_kfactor)
     ]
 
+    tasks = zip(definitions, [choose_worker()
+                              for _ in xrange(len(definitions))])
+
     return [(mapshed_start.s(opname, data).set(exchange=exchange,
                                                routing_key=geop_worker) |
              mapshed_finish.s().set(exchange=exchange,
                                     routing_key=geop_worker) |
              callback.s().set(link_error=errback,
                               exchange=exchange,
                               routing_key=choose_worker()))
-            for (opname, data, callback) in definitions]
+            for ((opname, data, callback), geop_worker) in tasks]
```

Comment:
Yes, that seemed to be the least disruptive way to do it. I was also impressed by how slight the change was!